Initialize viewer point at data center

grlee77 opened the pull request:
# Description

closes #3559

This PR replaces #3574 (it is updated based on recent changes to main and now focuses only on changing the viewer model's initial point to be at the midpoints of the LayerList's range). The change to include the point widths in the extent for Points layers will be added in a follow up PR.

Note that the increases to 'calls' in `test_dask_layers.py` are now less than in #3574.

<!-- What does this pull request (PR) do? Why is it necessary? -->
<!-- Tell us about your new feature, improvement, or fix! -->
<!-- If your change includes user interface changes, please add an image, or an animation "An image is worth a thousand words!" -->
<!-- You can use https://www.cockos.com/licecap/ or similar to create animations -->

## Type of change
<!-- Please delete options that are not relevant. -->
- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
<!-- What resources, documentation, and guides were used in the creation of this PR? -->
<!-- If this is a bug-fix or otherwise resolves an issue, reference it here with "closes #(issue)" -->

# How has this been tested?
<!-- Please describe the tests that you ran to verify your changes. -->
- [x] existing tests pass after updating the initial point position


## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

grlee77 committed:
initialize viewer_model point to midpoint of the data

jni commented on napari/layers/_tests/test_dask_layers.py:
```diff
@@ -184,7 +186,7 @@ def test_dask_unoptimized_slicing(delayed_dask_stack, monkeypatch):
     # all told, we have ~2x as many calls as the optimized version above.
     # (should be exactly 8 calls, but for some reason, sometimes less on CI)
-    assert delayed_dask_stack['calls'] >= 7
+    assert delayed_dask_stack['calls'] >= 10
```
Can you update the comment above about the expected number of calls?

grlee77 replied:
Done. also updated to set points relative to the initial point (which is no longer 0 since we start at the midpoint)

jni reviewed (approved):
Looks good! Just left a minor comment.

grlee77 committed:
reduce shear magnitude in test_warning_affine_slicing

this was necessary to maintain the same code path with two _slice_indices calls

grlee77 committed:
update logic of test_dask_global_optimized_slicing and test_dask_unoptimized_slicing

now use offsets relative to the initial point
formerly the initial point was always 0, but this is not the case now
that we initialize at the midpoint

grlee77 committed:
update point to midpoint in test_process_mouse_event

sofroniewn reviewed (approved):
Thanks for fixes, this is now good to go!
